File: solutions/csharp/food-chain/1/FoodChain.cs
using System.Text;

public static class FoodChain
{
    private static string[] _animal = { "", "fly", "spider", "bird", "cat", "dog", "goat", "cow", "horse" };
    private static string[] _animalComment =
    {
        "",
        "I don't know why she swallowed the fly. Perhaps she'll die.",
        "It wriggled and jiggled and tickled inside her.\n",
        "How absurd to swallow a bird!\n",
        "Imagine that, to swallow a cat!\n",
        "What a hog, to swallow a dog!\n",
        "Just opened her throat and swallowed a goat!\n",
        "I don't know how she swallowed a cow!\n",
        "She's dead, of course!",
    };
    private const int FIRST_VERSE = 1;
    private const int LAST_VERSE = 8;
    private const int SPIDER_VERSE = 3;

    public static string Recite(int verseNumber)
    {
        StringBuilder stringBuilder = new($"I know an old lady who swallowed a {_animal[verseNumber]}.\n");
        stringBuilder.Append(_animalComment[verseNumber]);

        if (verseNumber == FIRST_VERSE || verseNumber == LAST_VERSE)
            return stringBuilder.ToString();

        while (verseNumber > FIRST_VERSE)
        {
            if (verseNumber == SPIDER_VERSE)
                stringBuilder.Append("She swallowed the bird to catch the spider that wriggled and jiggled and tickled inside her.\n");
            else
                stringBuilder.Append($"She swallowed the {_animal[verseNumber]} to catch the {_animal[verseNumber - 1]}.\n");

            verseNumber--;
        }

        stringBuilder.Append(_animalComment[FIRST_VERSE]);
        return stringBuilder.ToString();
    }

    public static string Recite(int startVerse, int endVerse)
    {
        StringBuilder stringBuilder = new();

        int currentVerse = startVerse;
        while (currentVerse <= endVerse)
        {
            stringBuilder.Append(Recite(currentVerse));
            if (currentVerse != endVerse)
                stringBuilder.Append("\n\n");

            currentVerse++;
        }
        return stringBuilder.ToString();
    }
}

File: solutions/cpp/food-chain/1/food_chain.cpp
#include "food_chain.h"
#include <string>

namespace food_chain {
std::string verse(size_t verse_number) {
  std::string song = "I know an old lady who swallowed a ";
  song += food_info::animal.at(verse_number) + '.' + '\n';
  song += food_info::animal_adj.at(verse_number) + '\n';

  if (verse_number == food_info::LAST_VERSE ||
      verse_number == food_info::FIRST_VERSE) {
    return song;
  }
  for (; verse_number > food_info::FIRST_VERSE; --verse_number) {
    song += "She swallowed the " + food_info::animal.at(verse_number) +
            " to catch the " + food_info::animal.at(verse_number - 1);
    if (verse_number != food_info::SPIDER_VERSE) {
      song += ".\n";
    } else {
      song += " that wriggled and jiggled and tickled inside her.\n";
    }
  }
  song += food_info::animal_adj.at(food_info::FIRST_VERSE) + '\n';
  return song;
}
std::string verses(size_t start_verse_num, size_t end_verse_num) {
  std::string song{};
  for (size_t curr_verse = start_verse_num; curr_verse <= end_verse_num;
       ++curr_verse) {
    song += verse(curr_verse) + '\n';
  }
  return song;
}
std::string sing() { return verses(1, 8); }

} // namespace food_chain
